From: Haavard Skinnemoen <hskinnemoen@atmel.com>
Subject: [PATCH 2.6.22.atmel.5] Accept both rootwait and rootwait=1 options

The rootwait patch merged into 2.6.23 differs from the one we
backported to 2.6.22 in that the latter expects a rootwait=0/1
parameter while the former simply expects "rootwait".

This patch makes it possible to specify "rootwait" without a parameter
on 2.6.22 as well. If a parameter is given, a deprecation warning is
printed since this is incompatible with mainline.

Signed-off-by: Haavard Skinnemoen <hskinnemoen@atmel.com>
---
 init/do_mounts.c |   11 +++++++++--
 1 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/init/do_mounts.c b/init/do_mounts.c
index efc134c..a268de9 100644
--- a/init/do_mounts.c
+++ b/init/do_mounts.c
@@ -219,11 +219,18 @@ __setup("root=", root_dev_setup);
 
 static int __init rootwait_setup(char *line)
 {
-	root_wait = simple_strtol(line,NULL,0);
+	if (*line && *line != '=')
+		return 0;
+
+	if (*line)
+		printk(KERN_WARNING
+			"WARNING: \"rootwait=1\" is deprecated, "
+			"use \"rootwait\" instead.\n");
+	root_wait = 1;
 	return 1;
 }
 
-__setup("rootwait=", rootwait_setup);
+__setup("rootwait", rootwait_setup);
 
 static char * __initdata root_mount_data;
 static int __init root_data_setup(char *str)
-- 
1.5.3.2
